evalml/tests/automl_tests/test_automl.py

```diff
@@ -5037,3 +5062,70 @@ def test_automl_chooses_engine(engine_choice, X_y_binary):
             automl = AutoMLSearch(
                 X_train=X, y_train=y, problem_type="binary", engine=engine_choice
             )
+
+
+@patch("evalml.pipelines.components.RandomForestClassifier.predict_proba")
+@patch("evalml.pipelines.components.RandomForestClassifier.predict")
+@patch("evalml.pipelines.components.ElasticNetClassifier.predict_proba")
+@patch("evalml.pipelines.components.ElasticNetClassifier.predict")
+def test_automl_enesmbler_allowed_component_graphs(
+    mock_en_predict,
+    mock_en_predict_proba,
+    mock_rf_predict,
+    mock_rf_predict_proba,
+    X_y_regression,
+    caplog,
+):
+    """
+    Test that ensures that pipelines that are defined as strings or classes are able to be ensembled.
+    """
+    X, y = X_y_regression
+    mock_en_predict_proba.return_value = np.ones(len(y))
+    mock_rf_predict_proba.return_value = np.ones(len(y))
+    mock_en_predict.return_value = np.ones(len(y))
+    mock_rf_predict.return_value = np.ones(len(y))
+    component_graphs = {
+        "Pipeline1": {
+            "Imputer": ["Imputer", "X", "y"],
+            "Log Transformer": ["Log Transformer", "X", "y"],
+            "RF": ["Random Forest Regressor", "Imputer.x", "Log Transformer.y"],
+        },
+        "Pipeline2": {
+            "Imputer": ["Imputer", "X", "y"],
+            "Log Transformer": ["Log Transformer", "X", "y"],
+            "EN": ["Elastic Net Regressor", "Imputer.x", "Log Transformer.y"],
+        },
+    }
+    automl = AutoMLSearch(
+        X,
+        y,
+        "regression",
+        allowed_component_graphs=component_graphs,
+        ensembling=True,
+        max_batches=4,
+    )
+    automl.search()
```

Comment:
From skimming the docstring for this mock, I saw [this line](https://github.com/alteryx/evalml/blob/27198716f31a0e1661b0d4733c960be0d94c1acf/evalml/tests/conftest.py#L1101): 
> but it may not be suitable for all tests, such as tests that patch Estimator.fit instead of Pipeline.fit

Since most of the tests are mocking `predict_proba` and `predict` from the Estimator, I'm assuming that we can't unless something has changed.